src/config/effect.rs

```diff
@@ -10,8 +10,36 @@ pub enum Effect {
 
 #[derive(Copy, Clone, Debug, Deserialize)]
 pub struct DelayConfig {
+    #[serde(default = "DelayConfig::default_level")]
     pub level: f32,
+    #[serde(default = "DelayConfig::default_delay")]
     pub delay_ms: DelayMs,
+    #[serde(default = "DelayConfig::default_delay_max")]
+    pub max_delay_ms: DelayMs,
+    #[serde(default = "DelayConfig::default_num")]
```

Comment:
These `serde(default)` tags make it so that these values are optional in the YAML files, but when the file is deserialized, will be populated using these functions so that we always have a value. (Also, this moves responsibility for determining the max delay into the configuration, which I think makes sense, since it determines the range of the slider.)

Reply:
Yep! I'm not sure how exactly it does it, but it tends to give you a sensibly-placed error message as well.